http/user_agent.ts

```diff
@@ -184,21 +175,18 @@ function mapper(
     let j = 0;
     let k = 0;
     while (j < matchers.length && !matches) {
-      if (!matchers[j]) {
-        break;
-      }
       matches = matchers[j++]!.exec(ua);
 
       if (matches) {
         for (const processor of processors) {
-          const match = matches[++k]!;
```

Comment:
There were cases where `match` is actually `undefined`. So appending `!` here is not ideal.